### PR TITLE
{Storage} Enable logging for track2 data plane operations

### PR DESCRIPTION
**Pull request by Juliehzl**
**Description**

In track2 SDK, we have to configure the client to enable logging. Otherwise cx could not see the rest info.
**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Storage

File: src/azure-cli/azure/cli/command_modules/storage/_client_factory.py [hunk 1]
```diff
@@ -196,16 +196,20 @@ def get_account_url(cli_ctx, account_name, service):
...
     t_blob_service = get_sdk(cli_ctx, ResourceType.DATA_STORAGE_BLOB,
                              '_blob_service_client#BlobServiceClient')
     connection_string = kwargs.pop('connection_string', None)
     account_name = kwargs.pop('account_name', None)
     account_key = kwargs.pop('account_key', None)
     token_credential = kwargs.pop('token_credential', None)
     sas_token = kwargs.pop('sas_token', None)
+
+    # Enable NetworkTraceLoggingPolicy which logs all headers (except Authorization) without being redacted
+    client_kwargs['logging_enable'] = True
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/storage/_client_factory.py:209]: The same line is repeated 3 times. Is it possible to extract a common util func?

**Review by evelyn-ys — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/storage/_client_factory.py (hunk 1 above)
Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/storage/_client_factory.py]: done

File: src/azure-cli/azure/cli/command_modules/storage/_client_factory.py [hunk 2]
```diff
@@ -196,16 +197,19 @@ def get_account_url(cli_ctx, account_name, service):
...
+    client_kwargs = {}
     t_blob_service = get_sdk(cli_ctx, ResourceType.DATA_STORAGE_BLOB,
                              '_blob_service_client#BlobServiceClient')
     connection_string = kwargs.pop('connection_string', None)
     account_name = kwargs.pop('account_name', None)
     account_key = kwargs.pop('account_key', None)
     token_credential = kwargs.pop('token_credential', None)
     sas_token = kwargs.pop('sas_token', None)
+
+    client_kwargs.update(_prepare_client_kwargs_track2(cli_ctx))
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/storage/_client_factory.py:209]: We can simply use `client_kwargs = _prepare_client_kwargs_track2(cli_ctx)` if there is no modification before this line.

**Review by jiasli — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/storage/_client_factory.py [hunk 3]
```diff
@@ -3,7 +3,8 @@
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
 
-from azure.cli.core.commands.client_factory import get_mgmt_service_client, get_data_service_client
+from azure.cli.core.commands.client_factory import get_mgmt_service_client, get_data_service_client, \
+    _prepare_client_kwargs_track2
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/storage/_client_factory.py:7]: Good to know `_prepare_client_kwargs_track2` is useful for date-plane SDKs.

File: src/azure-cli/azure/cli/command_modules/storage/_client_factory.py (hunk 2 above)
Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/storage/_client_factory.py]: good catch. updated

**Review by evelyn-ys — COMMENTED** [review]
Is this pr duplicated with #16474

**Comment by Juliehzl** [comment]
> Is this pr duplicated with #16474

No. They are for different kwargs.

**Pull request merged**